### DISCUSSION TITLE
Restructure crate

### DISCUSSION THREAD

**Discussion by pleshevskiy** [pleshevskiy/estring#19] [Announcements]
Initially, I concentrated everything in the `core' module. The idea has matured a bit, and I want to distribute it into separate modules.

- `core` - will contain EString
- `std` - will contain implementations for standard types (`bool`, numbers, `Option`, etc.)
- `struct` - will contain predefined types (`SepVec`, `Pair`, `Triple`, etc.)
- `agg` - will contain aggregate functions (`Sum`, `Product`, etc)
- `low` - will contain low-level structures for more accurate string parsing

**Comment by brxken128** [comment]
I think this would be a good way to go about it.

**Reply by pleshevskiy** [reply]
I cannot use `struct`, because it's a reserved keyword. Can you suggest another name for this scope (predefined types (SepVec, Pair, Triple, etc.))?

**Reply by pleshevskiy** [reply]
nevermind) I renamed to `structs`

**Reply by brxken128** [reply]
> nevermind) I renamed to `structs`

I thought this would be an issue! I used `structs` too, it's close enough :)

**Comment by pleshevskiy** [comment]
I forgot about aggregates! I have a plan to move aggregates to the `agg` module.

**Comment by pleshevskiy** [comment]
It has been restructured in #21
